scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -0,0 +1,50 @@
+package com.twitter.scalding.beam_backend
+
+import com.stripe.dagon.Rule
+import com.twitter.scalding.Execution.{ToWrite, Writer}
+import com.twitter.scalding.typed._
+import com.twitter.scalding.{CFuture, Config, Execution, ExecutionCounters}
+import org.apache.beam.sdk.Pipeline
+import scala.concurrent.{ExecutionContext, Future}
+
+class BeamWriter(val beamMode: BeamMode) extends Writer {
+  override def start(): Unit = ()
+  override def finished(): Unit = ()
+
+  def getForced[T](conf: Config, initial: TypedPipe[T])(implicit cec: ExecutionContext): Future[TypedPipe[T]] = ???
+
+  def getIterable[T](conf: Config, initial: TypedPipe[T])(implicit cec: ExecutionContext): Future[Iterable[T]] = ???
+
+  override def execute(
+    conf: Config,
+    writes: List[ToWrite[_]])(implicit cec: ExecutionContext): CFuture[(Long, ExecutionCounters)] = {
+    import Execution.ToWrite._
+    val planner = BeamPlanner.plan(conf, beamMode.sources)
+    val phases: Seq[Rule[TypedPipe]] = OptimizationRules.standardMapReduceRules
```

Comment:
you might want to add a TODO to allow overriding these optimization rules based on the conf. This is done in cascading mode.